src/services/blockwatcher/service.rs

```diff
@@ -502,6 +528,9 @@ pub async fn process_new_blocks<
 	// Wait for both pipeline stages to complete
 	let (_process_result, _trigger_result) = tokio::join!(process_handle, trigger_handle);
 
+	// Clear fetched blocks to free memory for next iteration
+	block_tracker.clear_fetched_blocks(&network.slug).await;
```

Comment:
What happens if there's overlap in processing? E.g. the cron is set to 60 seconds, but a cycle takes 70 seconds to complete due to slow/delay in rpc. Next cycle overlaps with already running cycle. Now first cycle completed and cleared memory midsts second cycle.